### ISSUE TITLE
Fix netdata-updater for static installs

### ISSUE THREAD

**Issue by prologic** [netdata/netdata#8782]
This is a reoccuring bug that was fixed quite some time ago in #7197 then refixed and now it has come back. I must have missed something blindly obvious :/ This issue is to track and fix that once and for all.

**Comment by prologic** [comment]
I confirm this **does** happen:

```#!sh
root@vz3:~# ls -lah /etc/cron.daily/
total 60K
drwxr-xr-x   2 root root 4.0K Apr 21 07:37 .
drwxr-xr-x 146 root root  12K Mar 25 17:10 ..
-rwxr-xr-x   1 root root  311 May 29  2017 0anacron
-rwxr-xr-x   1 root root 1.5K May 28  2019 apt-compat
-rwxr-xr-x   1 root root  355 Oct 25  2016 bsdmainutils
-rwxr-xr-x   1 root root  384 Dec 12  2012 cracklib-runtime
-rwxr-xr-x   1 root root 1.2K Apr 18  2019 dpkg
-rwxr-xr-x   1 root root 4.1K Sep 27  2019 exim4-base
-rwxr-xr-x   1 root root  377 Aug 28  2018 logrotate
-rwxr-xr-x   1 root root 1.1K Feb 10  2019 man-db
-rwxr-xr-x   1 root root  249 May 17  2017 passwd
-rw-r--r--   1 root root  102 Oct  7  2017 .placeholder
root@vz3:~# find /opt/netdata | grep updater
/opt/netdata/usr/libexec/netdata/netdata-updater.sh
root@vz3:~# grep -E '\-\-accept' /opt/netdata/usr/libexec/netdata/netdata-updater.sh
  if sh "${TMPDIR}/netdata-latest.gz.run" --accept -- "${REINSTALL_OPTIONS}"; then
root@vz3:~#
```

I _actually_ don't get it :/ There should be no reason why this should be hapenning after my fix :/

**Comment by prologic** [comment]
I think I found the bug:

```
Unknown option ' --auto-update'. Ignoring it.
```

**Comment by prologic** [comment]
I have finally solved this! (cc @Kramerican). Really sorry about this mess! This has been a long standing bug that was attempted to be fixed twice. Turns out my fix wasn't quite complete and there were in fact two other bugs:

- Running the makeself archive with `sh` basically making all Bash(isms) __NOT WORK__ :/
- Not passing the `REINSTALL_OPTIONS` correctly as an array or word-splitting correctly.

This is really bad on our part and I sincerely apologize. In an effort to make this better I also fixed the fact that we had no real good ways of testing changes to the `netdata-updater.sh` script; so now its both possible to use a custom `NETDATA_NIGHTLIES_BASEURL` for dev/testing and if you want to use a custom URL.

I'll have the PR up tomorrow!

**Comment by DougParker1992** [comment]
I noticed the `netdata-updater` script was being removed from `/etc/cron.daily` from installations using `kickstart-static64.sh`. Will your PR fix this issue so that NetData will start automatically updating itself again?

**Comment by prologic** [comment]
> I noticed the `netdata-updater` script was being removed from `/etc/cron.daily` from installations using `kickstart-static64.sh`. Will your PR fix this issue so that NetData will start automatically updating itself again?

Yeap. I've been testing this on my Hypervisor ndoes where I install NetData via `kickstart-static64.sh`. I observe the same behavior of updates removes the cron entry and the "invalid option" message.

WIth my fix and improvement so I can test things locally without reaching out to our Google Storage bucket,things are working much better.

So yes. 😀

**Comment by DougParker1992** [comment]
Awesome! I'm looking forward to it. 😄

**Comment by prologic** [comment]
One thing though that's making me hesitant to put this PR up this morning is:

> Running the makeself archive with sh basically making all Bash(isms) NOT WORK :/

I cannot actually figure out what is going on here. From my macOS dev machine I cannot reproduce what I'm seeing on a Debian 10 system with NetData installed with `kickstart-static64.sh`.

On my macOS machine:

```#!sh
$ command -v sh
/bin/sh
$ realpath /bin/sh
/bin/sh
$ ls -lah /bin/sh
-rwxr-xr-x 1 root wheel 31K Apr  7 06:39 /bin/sh
```

On my Debian 10 with NetData (_wher eI'm testing this fix_):

```#!sh
root@vz1:~# command -v sh
/bin/sh
root@vz1:~# realpath /bin/sh
/bin/dash
root@vz1:~# ls -lah /bin/sh
lrwxrwxrwx 1 root root 4 Jan 18  2019 /bin/sh -> dash
```

So I "don't get it". Is `/bin/sh` that is actually a symlink to `/bin/bash` but if a script is exec'd with `sh` as `$0` does Bash enable POSIX compatibility mode? 🤔

**Comment by prologic** [comment]
And on a macOS machine `/bin/sh` is actually just a very old Bash:

```#!sh
$ sh --version
sh --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin19)
Copyright (C) 2007 Free Software Foundation, Inc.
```

**Comment by prologic** [comment]
PR is up #8808 Unfortunately I don't really have a way for anyone outside of NetData org to actually test this unless you are willing to go through what I did in the above "Test Plan". Let me know if you are and I would be more than willing to assist!

**Comment by cosmix** [comment]
@Ferroin please review and escalate to @netdata/core as needed.

**Comment by prologic** [comment]
As discussed, this ig going into Backlog and will be re-worked to have CI that verifies this works continuously which will also make review reasier and prevent future regressions of the updater.

**Comment by Simonmicro** [comment]
Hi, I wonder when this will be fixed. Netdata has a ton of daily installations (![badge](https://registry.my-netdata.io/api/v1/badge.svg?chart=web_log_nginx.requests_per_url&options=unaligned&dimensions=kickstart&group=sum&after=-86400&label=today&units=installations&precision=0)) and this breaks the update function for every static install of them (and I can imagine that this is also a huge percentage of them). Also about eleven instances managed by me have that problem and it is pretty frustrating to deal with them on a regular basis.

**Comment by DougParker1992** [comment]
This is an issue for me too. I have 7-10 instances that have to be manually updated and it's a big pain.

**Comment by prologic** [comment]
Relevant PR that fixes this is #8808
